Show procedures

glutouny opened the pull request:
This PR mainly adds the Show Procedures feature:
1. This query SQL is SHOW PROCEDURES
2. The result of this query is as follows：
![image](https://github.com/user-attachments/assets/81566822-278a-407e-bed3-5bd3ed23f291)

Commit:
Add SHOW PROCEDURES syntax to tree model

Commit:
Add SHOW PROCEDURES syntax to Table model

Commit:
Add SHOW PROCEDURES with ClassName

Commit:
Merge branch 'master' into show_procedures

glutouny commented:
This query SQL is SHOW PROCUDURES

jt2594838 commented:
It is a great job. 
I want to add two more columns in the result:
STAGE:  the detailed state of the procedure, for example,

Commit:
1. rename the State column to Status,
2. Add two more columns to the result:State、Progress

jt2594838 commented:
Could you please provide a screenshot with the newest columns?

glutouny commented:
<img width="3300" height="1090" alt="image" src="https://github.com/user-attachments/assets/1635b39a-b027-46c1-bf21-0c87b4794615" />
@jt2594838  this is  a screenshot with the newest columns.

jt2594838 commented:
One last thing, could you add an integration test for this?
Maybe you can start a thread creating timeseries constantly, until another thread sees any of them via SHOW PROCEDURES.

liyuheng55555 commented:
Hello @glutouny, glad to see this PR! I’ve just started reviewing it. 

So far, I feel that the “ClassName” column doesn’t need to use the full path—just keeping the final class name is enough, such as `CreateRegionGroupsProcedure` instead of `org.apache....CreateRegionGroupsProcedure`

liyuheng55555 commented on iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/execution/config/TableConfigTaskVisitor.java:
```diff
@@ -453,6 +455,14 @@ protected IConfigTask visitShowAINodes(
   }
 
+  @Override
+  protected IConfigTask visitShowProcedures(
+      final ShowProcedures showProceduresStatement, final MPPQueryContext context) {
+    context.setQueryType(QueryType.READ);
+    accessControl.checkUserIsAdmin(context.getSession().getUserName());
+    return new ShowProcedureTask();
+  }
```
Other SHOW statements are generally use the MAINTAIN privilege, so I suggest referring to that here.

liyuheng55555 replied:
Besides, the privilege control of new SQL needs to add IT, take a look at the function `testClusterManagementSql`

liyuheng55555 reviewed (changes requested):
PTAL

jt2594838 commented:
Well,  putting "Status" and "State" together may be confusing, so we'd better change "State" to "Phase".
And, please change "LastUpdate" to "LastUpdateTime".

Commit:
1. add an integration test[IoTDBProcedureIT.showProceduresTest] for "show procedures"
2. Add a function  toTProcedureInfo() to Procedure.java
3. move this define [showProcedures]  in the IoTDBSqlParser.g4 file to line 64
4. take a look at the function[show procedures] testClusterManagementSql
5. change "State" to "Phase". change "LastUpdate" to "LastUpdateTime".

glutouny commented:
this is a screenshot with the newest columns.
<img width="1914" height="232" alt="image" src="https://github.com/user-attachments/assets/4193d3d7-633d-440a-85e3-4fece0361298" />

Commit:
Merge branch 'master' into show_procedures

# Conflicts:
#	iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/sql/ast/AstVisitor.java

Commit:
Change "State" to "Phase" and resolve the conflict.

Commit:
Adjust the result columnHeader of show procedures

glutouny commented:
<img width="1918" height="194" alt="image" src="https://github.com/user-attachments/assets/f9d69f70-5198-4ca5-a7cb-db2b5596298e" />
Change "State" to "Phase"， this is a screenshot with the newest columns.

Commit:
Fix the bug where the thread for showProceduresTest is not enabled

Commit:
Modify the IT test IoTDBProcedureIT.showProceduresTest: createTimeseriesThread -> setTtlThread,
stopInserting -> stopSettingTTL
